File: server/src/models/CoinModel.js
const mongoose = require('mongoose');

const CoinSchema = new mongoose.Schema({
  name: { type: String, required: true },
  symbol: { type: String, required: true },
  telegramGroup: { type: String },
  twitterHandle: { type: String },
  currentPrice: { type: Number },
  marketCap: { type: Number },
  sentiment: {
    score: { type: Number, default: 0 },
    positiveCount: { type: Number, default: 0 },
    negativeCount: { type: Number, default: 0 }
  },
  lastAnalyzed: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Coin', CoinSchema);

File: flask/app/models/coin_model.py
from app import db
from sqlalchemy.sql import func

class CoinModel(db.Model):
    __tablename__ = 'coins'

    id = db.Column(db.Integer, primary_key=True)
    name = db.Column(db.String(100), nullable=False)
    symbol = db.Column(db.String(20), nullable=False, unique=True)
    telegram_group = db.Column(db.String(200))
    twitter_handle = db.Column(db.String(100))
    
    current_price = db.Column(db.Float)
    market_cap = db.Column(db.Float)
    
    sentiment_score = db.Column(db.Float, default=0)
    positive_mentions = db.Column(db.Integer, default=0)
    negative_mentions = db.Column(db.Integer, default=0)
    
    last_analyzed = db.Column(db.DateTime, server_default=func.now())
